bacs/src/main/java/com/adyen/checkout/bacs/BacsDirectDebitViewProvider.kt

```diff
@@ -22,13 +23,18 @@ internal object BacsViewProvider : ViewProvider {
         defStyleAttr: Int
     ): ComponentView {
         return when (viewType) {
-            BacsComponentViewType.INPUT -> BacsDirectDebitInputView(context, attrs, defStyleAttr)
-            BacsComponentViewType.CONFIRMATION -> BacsDirectDebitConfirmationView(context, attrs, defStyleAttr)
+            is BacsComponentViewType.Input -> BacsDirectDebitInputView(context, attrs, defStyleAttr)
+            is BacsComponentViewType.Confirmation -> BacsDirectDebitConfirmationView(context, attrs, defStyleAttr)
             else -> throw IllegalArgumentException("Unsupported view type")
         }
     }
 }
 
-internal enum class BacsComponentViewType : ComponentViewType {
-    INPUT, CONFIRMATION
+internal sealed class BacsComponentViewType(override val buttonTextResId: Int) : ButtonComponentViewType {
+    data class Input(
+        override val buttonTextResId: Int = R.string.bacs_continue
```

Comment:
Why would we need to supply another value then the default value here? If there is no good reason, I would opt to keep it an enum:
```Kotlin
internal enum class BacsComponentViewType(override val buttonTextResId: Int) : ComponentViewType {
    INPUT(1), CONFIRMATION(2)
}
```